Mathlib/Geometry/Synthetic/Avigad/Axioms.lean

```diff
@@ -0,0 +1,237 @@
+/-
+Copyright (c) 2023 André Hernandez-Espiet. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: André Hernandez-Espiet
+-/
+import Mathlib.Data.Real.Basic
+import Mathlib.Data.Set.Finite
+
+/-!
+Axioms of synthetic geometry
+-/
+
+/-! Universes for points lines and circles-/
+universe u
+
+/-- `IncidenceGeometry` represents geometry in the Euclidean sense, with primitives for points
+lines and circles-/
+class IncidenceGeometry where
+/--Points in the plane-/
+Point : Type u
+/--Lines in the plane-/
+Line : Type u
+/--Circles in the plane-/
+Circle : Type u
+
+/--A Point being on a line-/
+OnLine : Point → Line → Prop
+/--Two points being on the sameside of a line-/
+SameSide : Point → Point → Line → Prop
+/--Three points being in a row-/
+B : Point → Point → Point → Prop
+/--A point being the center of a center-/
+CenterCircle : Point → Circle → Prop
+/--A point being on a circle-/
+OnCircle : Point → Circle → Prop
+/--A point being inside a circle-/
+InCircle : Point → Circle → Prop
+/--That two lines intersect-/
+LinesInter : Line → Line → Prop
+/--A line and a circle intersect-/
+LineCircleInter : Line → Circle → Prop
+/--Two circles intersect-/
+CirclesInter : Circle → Circle → Prop
+/--The distance between two points-/
+length : Point → Point → ℝ
+/--The angle made between three points-/
+angle : Point → Point → Point → ℝ
+/--The constant dedicated to the right angle-/
+rightangle : ℝ
+/--The area made by three points-/
+area : Point → Point → Point → ℝ
+
+/--From a set of points getting one additional one-/
+more_pts : ∀ (S : Set Point), S.Finite → ∃ a, a ∉ S
+/--Interpolating a segment by an arbitrary amount-/
+pt_B_of_ne : ∀ {b c}, b ≠ c → ∃ a, B b a c
+/--Extending a segment by an arbitrary amount-/
+pt_extension_of_ne : ∀ {b c}, b ≠ c → ∃ a, B b c a
+/--Obtaining a point opposite a line and point-/
+diffside_of_not_online : ∀ {L a}, ¬OnLine a L → ∃ b, ¬OnLine b L ∧ ¬SameSide a b L
+/--Get a line from two points-/
+line_of_pts : ∀ a b, ∃ L, OnLine a L ∧ OnLine b L
+/--Getting a circle with center and point on it-/
+circle_of_ne : ∀ {a b}, a ≠ b → ∃ α, CenterCircle a α ∧ OnCircle b α
+/--If lines intersect then this gives you the point of intersection-/
+pt_of_linesinter : ∀ {L M}, LinesInter L M → ∃ a, OnLine a L ∧ OnLine a M
```

Comment:
```suggestion
pt_of_linesInter : ∀ {L M}, LinesInter L M → ∃ a, OnLine a L ∧ OnLine a M
```

If a theorem is about `FooBar x`, it should have `fooBar` in its name not `foobar`. I think the naming guide explains this a little better than I have.

Reply:
I see... Should the theorem then be `Point_of_LinesInter` instead?

Reply:
No, they should be `point_of_lineInter`; the first letter is kept lowercase. It's somewhat arbitrary, but it's what the style guidelines say, and following them makes things consistent.

Reply:
done I think?